### ISSUE TITLE
[RFC] Recommend to use Event/ instead of EventListener/

### ISSUE THREAD

**Issue by javiereguiluz** [symfony/symfony#23697]
| Q                | A
| ---------------- | -----
| Bug report?      | no
| Feature request? | no
| BC Break report? | no
| RFC?             | yes
| Symfony version  | 3.4

In the Symfony Docs and our default code, we use the `EventListener/` folder. This always looked "not OK" to me because it also contained event subscribers. When using Symfony Flex and Symfony 4, we're recommending to *only* use event subscribers and not event listeners, so the folder name is even worse.

Should we finally rename `EventListener/` to `Event/`? (This would also be consistent with the naming of the other folders: Command, Controller, Entity, Repository, etc.)

**Comment by Simperfit** [comment]
I'm really +1 on this.

Maybe we could go further maybe by telling people to name their event without adding EventSubscriber at the end, WDYT @javiereguiluz ?

**Comment by ogizanagi** [comment]
> In the Symfony Docs and our default code, we use the EventListener/ folder. This always looked "not OK" to me because it also contained event subscribers.

In the core, we also name classes `*Listener` while it implements `EventSubscriberInterface`, so I think the distinction in naming between listeners and subscribers is not even really relevant.

Starting recommending using `Event` in docs looks okay to me, but is it really worth it doing it in the core by moving classes from `EventListener/` to `Event/` (and thus having to handle deprecations for each of them)?

**Comment by nicolas-grekas** [comment]
> worth it doing it in the core by moving classes

not IMHO :)

A subscriber *is* a listener, but a listener is not an event, so I'd be reluctant here (but you can ignore my personal pov, just saying :) )

**Comment by netsuo** [comment]
What would you call an actual custom Event class and where would you put it ? How would you distinguish between a custom event and a subscriber/listener ?

**Comment by fabpot** [comment]
As already said, a subscriber is a listener as well. I don't see any problems that need to be solved here.

**Comment by chalasr** [comment]
Same, subscribers are listeners and listeners aren't events, different responsibilities. Nothing to change to me.

**Comment by javiereguiluz** [comment]
I'm sorry but I messed things up. In the Symfony Demo app we had listeners and now subscribers ... but we didn't change the folder name, so we have subscribers in a folder called listeners. The solution is simple: rename the `EventListener/` folder to `EventSubscriber/` (as we do in the Symfony Docs: https://symfony.com/doc/current/event_dispatcher.html#events-subscriber).

So there's no need to change the default name of the `EventListner/` folder because there won't be an `EventListener/` folder. Closing then. Thanks for the discussion.
